### ISSUE TITLE
Using slivar for structural variants

### ISSUE THREAD

**Issue by Hywel-W** [brentp/slivar#70]
HI, I'm trying to use slivar to identify de novo SVs. I have a merged vcf containing 27 trio samples that have been jointly genotyped for SVs but when I try to run a simple de novo script I get the attached error (slivar_error_1.txt). Looking at the error it seems like it may be due to my data not having DP in the INFO field of the vcf? if so I was wondering if there was a way of overriding this filter? Also, how do I save the output from this analysis?
[Slivar_error_1.txt](https://github.com/brentp/slivar/files/5224808/Slivar_error_1.txt)

for reference, here is the slivar script:

vcf=/scratch/c.wpchjw/LKS_WGS_SV_Analysis/Slivar/H_merged.vcf.gz
ped=/scratch/c.wpchjw/LKS_WGS_SV_Analysis/Slivar/LKS_slivar_ped.txt

# example command
./slivar expr --js slivar-functions.js \
        --vcf $vcf --ped $ped \
        --family-expr "denovo:fam.every(segregating_denovo)" \
        --pass-only \
        > slivar_3_out.txt

**Comment by brentp** [comment]
Hi, that slivar-functions.js are mostly for snp/indel vcfs.
If you have trios, you can write your own --trio function like:
```
--trio "denovo:kid.het && mom.hom_ref && dad.hom_ref && ..."
```
where you can replace ... with stuff to enforce quality. maybe that's e.g. "dad.GQ > 10 && mom.GQ > 10 && kid.GQ > 10"

if you used [duphold](https://github.com/brentp/duphold), it could be (for INFO.SVTYPE == "DEL"): `kid.DHFFC < 0.7 && mom.DHFFC > 0.8 && dad.DHFFC > 0.8`

basically anything that gives you a way to filter on quality. this is harder in SVs than in SNP/indel. which SV caller did you use?

**Comment by Hywel-W** [comment]
Thanks for the advice, I used the trio function and got it to run. However, I’m not sure how best to review the output as I used a merged vcf input file and the output is ~40k lines including the data on every individual. Is there a way to see the de novos for each trio other than running each trio individually, for example a list of de novo coordinates for each offspring?

I need to take a closer look at the quality metrics that delly gives to see if there is a way of restricting the number of SVs that are input as I’m currently getting around 3000 potential de novos per individual, which is obviously way too many!

It’s a good start and the slivar runs super fast.

Hywel

**Comment by brentp** [comment]
make sure to use `--pass-only` to get only variants that meet any filter written (default of slivar is to annotate and therefore write all variants).
if you give a `--ped` with only 1 trio, it will subset the output to contain only variants for that trio.

you can use [samplot](https://github.com/ryanlayer/samplot) and [samplot vcf](https://github.com/ryanlayer/samplot#usage-1) to draw pngs of putative *de novo* variants. It understand pedigree files so it will draw the appropriate trio(s) for each event.

**Comment by Hywel-W** [comment]
That’s really helpful , I’ll give it a go, thanks.
